AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessor.swift

```diff
@@ -429,15 +429,43 @@ extension ItemListProcessor: AuthenticatorKeyCaptureDelegate {
         _ captureCoordinator: AnyCoordinator<AuthenticatorKeyCaptureRoute, AuthenticatorKeyCaptureEvent>,
         key: String
     ) {
-        let dismissAction = DismissAction(action: { [weak self] in
-            Task {
-                await self?.parseAndValidateAutomaticCaptureKey(key)
+        if services.appSettingsStore.hasSeenDefaultSaveOptionPrompt {
+            switch services.appSettingsStore.defaultSaveOption {
+            case .saveLocally:
+                captureCoordinator.navigate(to: .dismiss(parseKeyAndDismiss(key, sendToBitwarden: false)))
+            case .saveToBitwarden:
+                captureCoordinator.navigate(to: .dismiss(parseKeyAndDismiss(key, sendToBitwarden: true)))
+            case .none:
+                coordinator.showAlert(.determineScanSaveLocation(
+                    saveLocallyAction: { [weak self] in
+                        captureCoordinator.navigate(
+                            to: .dismiss(self?.parseKeyAndDismiss(key, sendToBitwarden: false))
+                        )
+                    }, sendToBitwardenAction: { [weak self] in
+                        captureCoordinator.navigate(
+                            to: .dismiss(self?.parseKeyAndDismiss(key, sendToBitwarden: true))
+                        )
+                    }
+                ))
             }
-        })
-        captureCoordinator.navigate(to: .dismiss(dismissAction))
+        } else {
+            coordinator.showAlert(.determineScanSaveLocation(
```

Comment:
🤔 This seems to imply that even if the user hasn't started syncing the two apps, they'll still get an alert on a scan asking them where they want to save the code? Am I missing something?

Reply:
Great catch! 💯  I've added a test for this case explicitly and fixed it in the latest.